pkg/operator/ceph/object/notification/controller.go

```diff
@@ -195,10 +214,13 @@ func (r *ReconcileNotifications) reconcile(request reconcile.Request) (reconcile
 			notification,
 		)
 		if err != nil {
+			r.recorder.Eventf(notification, kapiv1.EventTypeNormal, string(cephv1.ReconcileFailed), "Reconcile failed for %s with error %s", bnName, err.Error())
 			return reconcile.Result{}, errors.Wrapf(err, "failed to provision CephBucketNotification %q for ObjectBucketClaims %q", bnName, bucketName)
 		}
+		r.recorder.Eventf(notification, kapiv1.EventTypeNormal, string(cephv1.Provisioned), "Notification %q provisioned on bucket %q", bnName, bucketName)
 	}
 
+	r.recorder.Eventf(notification, kapiv1.EventTypeNormal, string(cephv1.ReconcileSucceeded), "Successfully reconciled %q", bnName)
```

Comment:
The `ReconcileSucceeded` and `ReconcileFailed` events are added automatically by the reusable `ReportReconcileResult`. See here: https://github.com/BlaineEXE/rook/blob/cab3ce34fd96dd196c7ab1ff4de0c8182b97b1ce/pkg/operator/ceph/object/controller.go#L177

Please use this.

Reply:
what about the case that I return a retry indication without an error? (for the case that the topic is not provisioned yet)

Reply:
That should be an error. Similarly to how a Pod gets a failure event if a Volume source doesn't exist. It helps the user understand what is wrong and correct it if it is their mistake (not creating the Topic).

Reply:
its was specifically requested not to be an error in the original PR

Reply:
also, what about the case that i failed to fetch the notification object itself?
how can i report an event on it?

Reply:
last thing. if we use the `EventReporter` i cannot use the `FakeRecorder` in the unit test

should I add a `FakeReporter` to k8sutil?

Reply:
> it was specifically requested not to be an error in the original PR

I thought I had asked for it to be an error, but I may have not been clear at the time.


> last thing. if we use the `EventReporter` i cannot use the `FakeRecorder` in the unit test
> 
> should I add a `FakeReporter` to k8sutil?

Have you rebased on the latest master? I removed the deduplicating stuff from `ReportReconcileResult` so it should work with the `recorder.EventRecorder` now.

Reply:
works fine after rebase!